### PR TITLE
Spiegel.de: support embedded spiegeltv content

### PR DESCRIPTION
**Pull request by runningbits**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

spiegel.de sometimes embeds content from spiegel.tv with URLs currently not supported by youtube-dl, for example http://www.spiegel.de/sptv/spiegeltv/spiegel-tv-ueber-russische-propaganda-a-1136559.html .

The first part of this pull request targets the Spiegeltv extractor so it ignores some stuff at the end of the embedding url (`/embed/?autoplay=true`).

The second commit enhances the Spiegel extractor so it can pass the embedding url to the Spiegeltv extractor. Part of that is also to ensure that SpiegelArticleIE does not match /sptv/ urls.

I have to say that I'm not really satisfied with the second commit as it feels clumsy to add all the exceptions for /sptv/ in path but in the end this might be just because the spiegel.de website/path structure is pretty complex.

### CODE REVIEW

File: youtube_dl/extractor/spiegel.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 
 class SpiegelIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?spiegel\.de/video/[^/]*-(?P<id>[0-9]+)(?:-embed|-iframe)?(?:\.html)?(?:#.*)?$'
+    _VALID_URL = r'https?://(?:www\.)?spiegel\.de/(?:(?:video)|(?:sptv/spiegeltv))/[^/]*-(?P<id>[0-9]+)(?:-embed|-iframe)?(?:\.html)?(?:#.*)?$'
```
Comment (dstftw) [inline youtube_dl/extractor/spiegel.py]: Groups around `video` and `sptv/spiegeltv` are superfluous.

File: youtube_dl/extractor/spiegeltv.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 
 class SpiegeltvIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?spiegel\.tv/(?:#/)?filme/(?P<id>[\-a-z0-9]+)'
+    _VALID_URL = r'https?://(?:www\.)?spiegel\.tv/(?:#/)?filme/(?P<id>[\-a-z0-9]+)(?:/embed/?)?(?:\?.+)?'
```
Comment (dstftw) [inline youtube_dl/extractor/spiegeltv.py]: This change does not make any sense. Old regex matches http://www.spiegel.tv/filme/putins-trollfabriken/embed/?autoplay=true just fine.

**Comment by runningbits** [comment]
pushed new version of commits with review comments (hopefully) addressed.

**Comment by FelixSchwarz** [comment]
This pull request is likely obsolete due to the site relaunch on 2017-05-10 (see #13159).

**Pull request closed without merging**